### DISCUSSION TITLE
How to clear an iteration or single-select field with `updateProjectV2ItemFieldValue`

### DISCUSSION THREAD

**Discussion by jtoar** [community/community#31127] [Projects and Issues]
Hey, I'm in the process of migrating from the ProjectNext API to the ProjectsV2 API and I can't seem to clear an iteration or single-select field's value anymore. Before, using the ProjectNext API, if I passed an empty string instead of an ID to an iteration or single-select field, it'd clear it:

```js
  await octokit.graphql(MUTATION, {
    projectId: '...',
    itemId: '...',
    fieldId: '...',
    value: '' // The field I'm updating here is a single select field, I and want to clear it completely.
              // As if it hadn't been set. Passing an empty string used to work.
  })
```

Where the mutation is:

```gql
  mutation UpdateProjectNextItemFieldMutation(
    $projectId: ID!
    $itemId: ID!
    $fieldId: ID!
    $value: String!
  ) {
    updateProjectNextItemField(
      input: {
        projectId: $projectId
        itemId: $itemId
        fieldId: $fieldId
        value: $value
      }
    ) {
      projectNextItem {
        id
      }
    }
  }
```

Now if I try that with the ProjectsV2 API, I have to change `value` to something like `value: { singleSelectOptionId: "" }`, but that doesn't clear it anymore. Instead it throws a validation error:

```
The single select option Id does not belong to the field
```

Is there any way to do this? Thanks!

**Comment by kpollich** [comment]
Having this exact issue right now as well. Having no way to provide an explicitly null/empty value via the GraphQL API is a real pain when developing API integrations against Projects V2. Is there a workaround for this? Sniffing around on the network tab for the projects UI, I can see that when making REST calls GitHub passes `""` for empty values, but this simply doesn't work in GraphQL currently.

**Comment by miriam-groeneveld** [comment]
You can clear iteration field and other fields (text, number, date, assignees, labels, single-select, and milestone fields) using [clearprojectv2itemfieldvalue](https://docs.github.com/en/graphql/reference/mutations#clearprojectv2itemfieldvalue)
